qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -363,27 +364,51 @@ def __setitem__(self, column, value):
             raise QiitaDBColumnError("Column %s does not exist in %s" %
                                      (column, self._dynamic_table))
 
+        conn_handler.add_to_queue(queue, sql, (value, self._id))
+
+    def __setitem__(self, column, value):
+        r"""Sets the metadata value for the category `column`
+
+        Parameters
+        ----------
+        column : str
+            The column to update
+        value : str
+            The value to set. This is expected to be a str on the assumption
+            that psycopg2 will cast as necessary when updating.
+
+        Raises
+        ------
+        ValueError
+            If the value type does not match the one in the DB
+        """
+        conn_handler = SQLConnectionHandler()
+        queue_name = "set_item_%s" % self._id
+        conn_handler.create_queue(queue_name)
+
+        self.add_setitem_queries(column, value, conn_handler, queue_name)
+
         try:
-            conn_handler.execute(sql, (value, self._id))
-        except Exception as e:
+            conn_handler.execute_queue(queue_name)
+        except QiitaDBExecutionError as e:
             # catching error so we can check if the error is due to different
             # column type or something else
-            column_type = conn_handler.execute_fetchone(
-                """SELECT data_type
-                   FROM information_schema.columns
-                   WHERE column_name=%s AND table_schema='qiita'
-                """, (column,))[0]
-            value_type = type(value).__name__
-
-            if column_type != value_type:
+            if "invalid input syntax for" in str(e):
```

Comment:
Something like [this](https://github.com/biocore/qiita/blob/master/qiita_db/search.py#L277) from search.py would probably be cleaner, since it directly connects the python datatype to the SQL datatype and the invalid syntax could be for other reasons besides type.